#!/usr/bin/env python
PACKAGE = "simple_local_path_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_angular_velocity_degrees", double_t,   0, "Minimum angular velocity in degrees", 5, 1, 50)
gen.add("max_angular_velocity_degrees", double_t,   0, "Maximum angular velocity in degrees", 40, 1, 50)
gen.add("min_linear_velocity",          double_t,   0, "Minimum linear velocity in m/s", 0.05,  0.01, 1)
gen.add("max_linear_velocity",          double_t,   0, "Maximum linear velocity in m/s", 0.5,  0.01, 1)
gen.add("angular_tolerance_degrees",    double_t,   0, "How close to target angle do we need to be to count as success", 5,  1, 10)
gen.add("linear_tolerance",             double_t,   0, "How close to target angle do we need to be to count as success", 0.05,  0.01, 0.1)
gen.add("collision_cost_threshold",     double_t,   0, "The cost map threshold at which collision is detected", 150,  50, 300)
gen.add("waypoint_step_size",           int_t,      0, "Number of waypoints to skip when moving setting next target", 10,  1, 20)

exit(gen.generate(PACKAGE, "simple_local_path_planner", "NodeParameters"))
